Update DISTRO as per new branding for Flex OS and remove external toolchains support

hashraf91 opened the pull request:
This also reverts commit bdd684f045a4b5c26014e70d26d2598ed735d806 which introduced intermediate changes.

This removes "Sokol" keyword from the repository except where the repository link is mentioned as we are not renaming the repository name yet.

This changes following DISTRO related overrides:
```
sokol-flex         -> flex-os
sokol-flex-common  -> flex-common
sokol-flex-distro  -> flex-distro
sokol-flex-staging -> flex-staging
sokol-flex-support -> flex-support
```

Remove sourcery and external toolchains support

Hence, removed TCMODE from local.conf and set to 'default' in flex-os.conf. Also removed TOOLCHAINS_PATH from local.conf and flex-os.conf as we don't install any toolchain through CodeBench installer anymore.

JIRA-ID: SB-23722

hashraf91 committed:
Update DISTRO as per new branding for Flex OS

This also reverts commit bdd684f045a4b5c26014e70d26d2598ed735d806
which introduced intermediate changes.

This removes "Sokol" keyword from the repository except where the
repository link is mentioned as we are not renaming the repository
name yet.

This changes following DISTRO related overrides:
sokol-flex         -> flex-os
sokol-flex-common  -> flex-common
sokol-flex-distro  -> flex-distro
sokol-flex-staging -> flex-staging
sokol-flex-support -> flex-support

JIRA-ID: SB-23722

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

Noor-Ahsan commented on meta-flex-distro/recipes-core/packagegroups/packagegroup-base.bbappend:
```diff
@@ -2,6 +2,6 @@
 # ---------------------------------------------------------------------------------------------------------------------
 
-FILESEXTRAPATHS:prepend:feature-sokol-flex-staging := "${THISDIR}/${BPN}:"
+RDEPENDS:packagegroup-base-bluetooth:append:flex-os = "${@bb.utils.contains('DISTRO_FEATURES', 'pulseaudio', 'pulseaudio pulseaudio-server', '', d)}"
```
This is not only the distro change. You are doing some cleanup as well. I think its explaination should be added

hashraf91 replied:
Nope, there is no other cleanup other than removal of Sokol keyword which is mentioned in the description. This diff is just because git is confusing file renaming of different files.

hashraf91 committed:
Remove sourcery and external toolchains support

We only support internal toolchain i.e. oe/yocto-built toolchain,
moving forward.

Hence, removed TCMODE from local.conf and set to 'default' in
flex-os.conf. Also removed TOOLCHAINS_PATH from local.conf and
flex-os.conf as we don't install any toolchain through CodeBench
installer anymore.

JIRA-ID: SB-23722

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

Noor-Ahsan commented:
Remove https://github.com/MentorEmbedded/meta-sokol-flex/blob/e4091609a5fd5117282af1756a5b717850e94db0/meta-flex-distro/conf/distro/flex-os.conf#L73

Noor-Ahsan commented:
rest looks good

hashraf91 commented:
> Remove
> 
> https://github.com/MentorEmbedded/meta-sokol-flex/blob/e4091609a5fd5117282af1756a5b717850e94db0/meta-flex-distro/conf/distro/flex-os.conf#L73

Updated.
